### PR TITLE
Minor code fixes to mars / response_file

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
Pretty trivial code changes

### CODE REVIEW

File: src/root/response.d [hunk 1]
```diff
@@ -42,24 +50,19 @@ import ddmd.root.file, ddmd.root.filename;
  *   0   success
  *   !=0   failure (argc, argv unchanged)
  */
-extern (C++) bool response_expand(Strings* args)
+bool response_expand(Strings* args)
 {
-    const(char)* cp;
     int recurse = 0;
-    for (size_t i = 0; i < args.dim;)
+    foreach (i, cp; *args)
```
Comment (WalterBright) [inline src/root/response.d]: Note line 61. `foreach` can behave erratically if `args` gets resized inside the loop.

File: src/gluelayer.d [hunk 2]
```diff
@@ -92,18 +92,18 @@ else
     alias elem = ddmd.backend.elem;
     alias type = ddmd.backend.type;
 
-    extern extern (C++) void obj_write_deferred(Library library);
-    extern extern (C++) void obj_start(char* srcfile);
-    extern extern (C++) void obj_end(Library library, File* objfile);
-    extern extern (C++) void genObjFile(Module m, bool multiobj);
+    extern (C++) void obj_write_deferred(Library library);
```
Comment (WalterBright) [inline src/gluelayer.d]: Just use `extern (C++) { ... }` for these declarations.

File: src/root/response.d (hunk 1 above)
Comment (mathias-lang-sociomantic) [inline src/root/response.d]: Good catch. I'll revert.

**Comment by mathias-lang-sociomantic** [comment]
Updated:
- Reverted `for` -> `foreach` in `root.response`;
- Nested all declarations in `extern (C++) {...}` in `gluelayer` (module-wise)

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Comment by yebblies** [comment]
Nice, those `static`s should never have gone in.

**Pull request merged**